artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -0,0 +1,155 @@
+import asyncio
+import concurrent
+import inspect
+import logging
+import os
+import typing
+
+from google.cloud import bigquery
+
+from artcommonlib import logutil
+from artcommonlib.konflux import konflux_build_record
+
+
+class KonfluxDb:
+    REQUIRED_VARS = ['GOOGLE_CLOUD_PROJECT', 'GOOGLE_APPLICATION_CREDENTIALS', 'DATASET_ID', 'TABLE_ID']
+
+    def __init__(self):
+        self._check_env_vars()
+        self.client = bigquery.Client()
+        self.dataset_id = os.environ['DATASET_ID']
+        self.table_ref = f'{self.client.project}.{self.dataset_id}.{os.environ["TABLE_ID"]}'
+        self.logger = logutil.get_logger(__name__)
+        self.column_names = self._get_column_names()  # e.g. "(name, group, version, release, ... )"
+
+        # Make the gcp logger less noisy
+        logging.getLogger('google.auth.transport.requests').setLevel(logging.WARNING)
+
+    def _check_env_vars(self):
+        """
+        Check if all required env vars are defined. Raise an exception otherwise
+        """
+
+        for var in self.REQUIRED_VARS:
+            if not os.getenv(var):
+                raise RuntimeError(f'Environment variable {var} must be defined')
```

Comment:
Here will be more convenient if we print all missing required env vars in one message.

Reply:
Done in 4ba453e0fb03aa8bb58c9b109d763db56dfcdde2